src/ProtectedRoute.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useRef } from 'react'
```

Comment:
It appears that this file is located in the wrong location. I recommend relocating it to the `component` folder.